ev3dev2/display.py

```diff
@@ -195,8 +196,21 @@ class Display(FbMem):
     def __init__(self, desc='Display'):
         FbMem.__init__(self)
 
+        self.platform = get_current_platform()
+
+        if self.platform == "ev3" and self.var_info.bits_per_pixel == 1:
+            # Pre 4.14 kernel
+            im_type = "1"
+        elif self.platform == "ev3" and self.var_info.bits_per_pixel == 32:
+            # Post 4.14 kernel
+            im_type = "L"
+        elif self.platform == "pistorms" and self.var_info.bits_per_pixel == 16:
```

Comment:
Please only check for the special case of platform on EV3. We use 16bpp on at least 3 different devices.